junhaoyu-aws/opentelemetry-collector-contrib
# s3mapprovider for Collector: load configuration from config files in Amazon S3

**Description:**

An implementation of ConfigMapProvider for Amazon S3 (s3mapprovider) allows OTEL Collector the ability to load configuration for itself by fetching and reading config files stored in Amazon S3.

**Link to tracking Issue:**
#12939

**Testing:**
First, I mocked how a S3 client works while interacting with a S3 server.
Second, I designed and passed all the unit tests for edge cases such as empty fields, invalid config files, non-exist config files and so on.
Third, I did integration tests such as:
- Start an OTEL Collector with s3mapprovider.
- Start an OTEL sample application to send telemetry data to check if the OTEL Collector is running well.

**Documentation:**
Added a README.md.

**Sponsor:**
@Aneurysm9
